src/main/java/com/kobylynskyi/graphql/codegen/kotlin/KotlinGraphQLTypeMapper.java

```diff
@@ -0,0 +1,151 @@
+package com.kobylynskyi.graphql.codegen.kotlin;
+
+import com.kobylynskyi.graphql.codegen.java.JavaGraphQLTypeMapper;
+import com.kobylynskyi.graphql.codegen.mapper.GraphQLTypeMapper;
+import com.kobylynskyi.graphql.codegen.mapper.ValueMapper;
+import com.kobylynskyi.graphql.codegen.model.MappingContext;
+import com.kobylynskyi.graphql.codegen.model.NamedDefinition;
+import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLOperation;
+import com.kobylynskyi.graphql.codegen.utils.Utils;
+import graphql.language.Argument;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static com.kobylynskyi.graphql.codegen.java.JavaGraphQLTypeMapper.JAVA_UTIL_LIST;
+import static java.util.Arrays.asList;
+
+/**
+ * @author 梦境迷离
+ * @since 2020/12/09
+ */
+public class KotlinGraphQLTypeMapper implements GraphQLTypeMapper {
+
+    private static final String KOTLIN_UTIL_LIST = "List";
+    private static final String KOTLIN_UTIL_OPTIONAL = "?";
+    // Char Boolean are not primitive type, but non null equivalent jvm primitive types.
+    private static final Set<String> KOTLIN_PRIMITIVE_TYPES = new HashSet<>(asList("Byte", "Short", "Int", "Long", "Float", "Double", "Char", "Boolean"));
+
+    private final ValueMapper valueMapper;
+
+    public KotlinGraphQLTypeMapper(ValueMapper valueMapper) {
+        this.valueMapper = valueMapper;
+    }
+
+    public static boolean isKotlinPrimitive(String scalaType) {
+        return KOTLIN_PRIMITIVE_TYPES.contains(scalaType);
+    }
+
+    /**
+     * Wrap Kotlin type into {@link List}.
+     * E.g.: {@code String} becomes {@code List<String>} in Kotlin
+     *
+     * @param type           The name of a type that will be wrapped into {@code List<>} in Java/Kotlin or {@code Seq[]} in Scala
+     * @param mappingContext Global mapping context
+     * @return String The name of the given type, wrapped into {@code List<>} in Java/Kotlin or {@code Seq[]} in Scala
+     */
+    @Override
+    public String wrapIntoList(MappingContext mappingContext, String type) {
+        return getGenericsString(mappingContext, KOTLIN_UTIL_LIST, type);
+    }
+
+    /**
+     * Return upper bounded wildcard for the given interface type:
+     * {@code "Foo"} becomes {@code "List<out Foo>"} in Kotlin.
+     *
+     * @param type           The name of a type whose upper bound wildcard will be wrapped into a list.
+     * @param mappingContext Global mapping context
+     * @return String The name of the the wrapped type.
+     */
+    @Override
+    public String wrapSuperTypeIntoList(MappingContext mappingContext, String type) {
+        return getGenericsString(mappingContext, KOTLIN_UTIL_LIST, "out " + type);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
```

Comment:
These doc comments are unnecessary. Javadoc is inherited by default. We only need `@inheritDoc` if we need to write other javadoc but still include the parent's doc as far as I remember.